### PR TITLE
Add generic grid view micro framework

### PR DESCRIPTION
**Pull request by mrylmz**

### CODE REVIEW

File: Modules/GridView/GridView.swift [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+
+        collectionView.backgroundColor = .clear
+        addSubview(collectionView)
+        collectionView.clipsToBounds = false
+        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(collectionView)
```
Comment (JensK611) [inline Modules/GridView/GridView.swift]: Why are you adding the collectionView multiple times as subview? I guess one call is superfluous.

File: Modules/GridView/GridView.swift [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    public func collectionView(
+        _ collectionView: UICollectionView,
+        layout collectionViewLayout: UICollectionViewLayout,
+        sizeForItemAt indexPath: IndexPath
+    ) -> CGSize {
+        let bounds = collectionView.bounds.inset(by: model.insets)
+        let totalColumnSpacing = model.spacing.width * CGFloat(max(0, model.numberOfColumns - 1))
+        let width = (bounds.width - totalColumnSpacing) / CGFloat(max(1, model.numberOfColumns))
+
+        return CGSize(width: width, height: width)
```
Comment (JensK611) [inline Modules/GridView/GridView.swift]: Maybe we can find a more suitable name for a parameter which is both `width` and `height`

File: Modules/GridView/GridViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import Foundation
+import JamitFoundation
+import UIKit
+
+// TODO: Write documentation for public interface!
```
Comment (JensK611) [inline Modules/GridView/GridViewModel.swift]: Please remove TODO and add documentation.

File: Modules/GridView/GridView.swift [hunk 4]
```diff
@@ -0,0 +1,97 @@
+import Foundation
+import JamitFoundation
+import UIKit
+
+// TODO: Write documentation for public interface!
```
[the remark this answers is not in the document]
Comment (mrylmz) [reply Modules/GridView/GridView.swift]: Should be added now please have a look on the documentation if it is descriptive enough.

File: Modules/GridView/GridView.swift [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        return model.items.count
+    }
+
+    public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeue(cellOfType: ItemViewCell.self, for: indexPath)
+        cell.model = model.items[indexPath.item]
+        return cell
+    }
+
+    // MARK: - UICollectionViewDelegate
```
[the remark this answers is not in the document]
Comment (mrylmz) [reply Modules/GridView/GridView.swift]: I have added it there to indicate that implementations of the `UICollectionViewDelegate` interface should be placed there.

File: Modules/GridView/GridView.swift (hunk 1 above)
Comment (mrylmz) [reply Modules/GridView/GridView.swift]: I think that was a mistake after making a git rebase :)

File: Modules/GridView/GridView.swift (hunk 2 above)
Comment (mrylmz) [reply Modules/GridView/GridView.swift]: How do you think about renaming it to `dimension`?

File: Modules/GridView/GridView.swift (hunk 5 above)
[the remark this answers is not in the document]
Comment (JensK611) [reply Modules/GridView/GridView.swift]: Ok fine 😅

File: Modules/GridView/GridView.swift (hunk 2 above)
Comment (JensK611) [reply Modules/GridView/GridView.swift]: Yes `dimension` would be great!

File: Modules/GridView/GridView.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (JensK611) [reply Modules/GridView/GridView.swift]: Documentation is good! Thanks!

**Pull request merged**